File: frontend/src/api/github.ts
import api from "axios";

export interface PushGithubRequest {
  language: string;
  code: string;
  name: string;   
}

export interface PushGithubResponse {
  path: string;
  sha: string;
  url: string;
}

export async function pushSolutionToGithub(
  questionId: number,
  request: PushGithubRequest,
) {
  const response = await api.post<PushGithubResponse>(
    `/api/questions/${questionId}/github`,
    request,
  );

  return response.data;
}

File: backend/internal/dto/github.go
package dto

type PushGithubRequest struct {
	Language string `json:"language" binding:"required"`
	Name     string `json:"name" binding:"required"`
	Code     string `json:"code" binding:"required"`
}

type PushGithubResponse struct {
	Path string `json:"path"`
	SHA  string `json:"sha"`
	URL  string `json:"url"`
}
